optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/Sequence.java

```diff
@@ -41,10 +45,32 @@ protected Sequence(ConsecutiveSetTree<ValueType_, ?, ?> sourceTree, NavigableSet
         this.count = count;
     }
 
-    public NavigableSet<ValueType_> getItems() {
+    public Collection<ValueType_> getItems() {
         return consecutiveItemsSet;
     }
 
+    public ValueType_ getFirstItem() {
+        if (firstItem == null) {
+            firstItem = consecutiveItemsSet.first();
+        }
+        return firstItem;
+    }
+
+    public ValueType_ getLastItem() {
+        if (lastItem == null) {
+            lastItem = consecutiveItemsSet.last();
+        }
+        return lastItem;
+    }
```

Comment:
I added these convenience methods, as I noticed this being used very frequently, even in the constraint provider.
The cache is not entirely important, but as long as it is lazy, I think it doesn't hurt.